src/plugins/discover/public/application/apps/main/components/doc_table/doc_table_infinite.tsx

```diff
@@ -16,25 +16,88 @@ import { SkipBottomButton } from '../skip_bottom_button';
 
-const DocTableInfiniteContent = (props: DocTableRenderProps) => {
-  const [limit, setLimit] = useState(props.minimumVisibleRows);
+const DocTableWrapperMemoized = memo(DocTableWrapper);
 
-  // Reset infinite scroll limit
-  useEffect(() => {
-    setLimit(props.minimumVisibleRows);
-  }, [props.rows, props.minimumVisibleRows]);
+interface DocTableInfiniteContentProps extends DocTableRenderProps {
+  limit: number;
+  onSetMaxLimit: () => void;
+  onBackToTop: () => void;
+}
+
+const DocTableInfiniteContent = ({
+  rows,
+  columnLength,
+  sampleSize,
+  limit,
+  onSkipBottomButtonClick,
+  renderHeader,
+  renderRows,
+  onSetMaxLimit,
+  onBackToTop,
+}: DocTableInfiniteContentProps) => {
+  const onSkipBottomButton = useCallback(() => {
+    onSetMaxLimit();
+    onSkipBottomButtonClick();
+  }, [onSetMaxLimit, onSkipBottomButtonClick]);
+
+  return (
+    <Fragment>
+      <SkipBottomButton onClick={onSkipBottomButton} />
+      <table className="kbn-table table" data-test-subj="docTable">
+        <thead>{renderHeader()}</thead>
+        <tbody>{renderRows(rows.slice(0, limit))}</tbody>
+        <tfoot>
+          <tr>
+            <td colSpan={(columnLength || 1) + 2} style={FOOTER_PADDING}>
+              {rows.length === sampleSize ? (
+                <div
+                  className="kbnDocTable__footer"
+                  data-test-subj="discoverDocTableFooter"
+                  tabIndex={-1}
+                  id="discoverBottomMarker"
+                >
+                  <FormattedMessage
+                    id="discover.howToSeeOtherMatchingDocumentsDescription"
+                    defaultMessage="These are the first {sampleSize} documents matching
+your search, refine your search to see others."
+                    values={{ sampleSize }}
+                  />
+                  <EuiButtonEmpty onClick={onBackToTop} data-test-subj="discoverBackToTop">
+                    <FormattedMessage
+                      id="discover.backToTopLinkText"
+                      defaultMessage="Back to top."
+                    />
+                  </EuiButtonEmpty>
+                </div>
+              ) : (
+                <span tabIndex={-1} id="discoverBottomMarker">
+                  &#8203;
+                </span>
+              )}
+            </td>
+          </tr>
+        </tfoot>
+      </table>
+    </Fragment>
+  );
+};
+
+export const DocTableInfinite = (props: DocTableProps) => {
+  const tableWrapperRef = useRef<HTMLDivElement>(null);
+  const [limit, setLimit] = useState(50);
 
   /**
    * depending on which version of Discover is displayed, different elements are scrolling
    * and have therefore to be considered for calculation of infinite scrolling
    */
   useEffect(() => {
-    const scrollDiv = document.querySelector('.kbnDocTableWrapper') as HTMLElement;
+    // After mounting table wrapper should be initialized
+    const scrollDesktopElem = tableWrapperRef.current as HTMLDivElement;
```

Comment:
Quick note about this change of infinite table, their refactoring was done because of the issue that paginated table could be displayed multiple times on a dashboard, right? The "infinite" table is just used on the main app so, there can't be multiple on. however I think this approach is cleaner not, just want to be sure I understood the reasoning for the refactoring of this part.

Reply:
Yes, it was refactored since I have added `forwardRef` to a wrapper. I thought It might be better to replace one `querySelector('.kbnDocTableWrapper')` with ref in infinite table too

Reply:
yes, it's cleaner that way, thx